libfreerdp/gdi/gdi.c

```diff
@@ -614,12 +631,25 @@ static void gdi_patblt(rdpContext* context, PATBLT_ORDER* patblt)
 				hatched, backColor, foreColor, gdi->palette);
 
 		hBmp = gdi_CreateBitmap(8, 8, gdi->drawing->hdc->bitsPerPixel, data);
+		if (!hBmp)
+		{
+			_aligned_free(data);
+			ret = FALSE;
+			goto out_error;
+		}
 
 		originalBrush = gdi->drawing->hdc->brush;
 		gdi->drawing->hdc->brush = gdi_CreateHatchBrush(hBmp);
+		if (!gdi->drawing->hdc->brush)
+		{
+			_aligned_free(data);
+			ret = FALSE;
+			goto out_error;
+		}
 
-		gdi_PatBlt(gdi->drawing->hdc, patblt->nLeftRect, patblt->nTopRect,
-		patblt->nWidth, patblt->nHeight, gdi_rop3_code(patblt->bRop));
+		if (gdi_PatBlt(gdi->drawing->hdc, patblt->nLeftRect, patblt->nTopRect,
+		patblt->nWidth, patblt->nHeight, gdi_rop3_code(patblt->bRop)) != 1)
```

Comment:
This is ok for freerdp's api but I'd suggest to check for == 0 to match the msdn description.